src/internal/internal_geset.cc

```diff
@@ -92,44 +92,20 @@ void set(internal::TargetType<Target::Devices>,
 {
     using ij_tuple = typename BaseMatrix<scalar_t>::ij_tuple;
 
-    int64_t mt = A.mt();
-    int64_t nt = A.nt();
-
-    // Find ranges of matching mb's.
-    std::vector< int64_t > irange;
-    int64_t last_mb = -1;
-    for (int64_t i = 0; i < mt; ++i) {
-        int64_t mb = A.tileMb( i );
-        if (mb != last_mb) {
-            last_mb = mb;
-            irange.push_back( i );
-        }
-    }
-    irange.push_back( mt );
-
-    // Find ranges of matching nb's.
-    std::vector< int64_t > jrange;
-    int last_nb = -1;
-    for (int64_t j = 0; j < nt; ++j) {
-        int64_t nb = A.tileNb( j );
-        if (nb != last_nb) {
-            last_nb = nb;
-            jrange.push_back( j );
-        }
-    }
-    jrange.push_back( nt );
+    // Find ranges of matching mb's and ranges of matching nb's.
+    std::vector< int64_t > irange = device_regions_range( true, A );
+    std::vector< int64_t > jrange = device_regions_range( false, A );
```

Comment:
Can we use an enum for the Row or Col, instead of a bool? Without memorizing the API, at this point I don't know what `true` and `false` means here. E.g., `RowCol::Row` and `RowCol::Col`. (Probably need to import that enum from LAPACK++ into SLATE in slate/include/enums.hh.)